src/bot.ts

```diff
@@ -1,14 +1,42 @@
-// Read config
-import * as config from './config';
-
 // Main logic
-import {FlameBot} from './FlameBot';
+import {MattermostFlameBot} from './services/mattermost';
+import {TelegramFlameBot} from './services/telegram';
 
 // Telegram integration
 import * as TelegramBot from 'node-telegram-bot-api';
 
-const flameRate = 0.02;
-const telegram = new TelegramBot(config.token, { polling: true });
-const flameBot = new FlameBot(flameRate, telegram);
+// Dotenv
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+// Configuration
+const telegramToken = process.env.TELEGRAM_TOKEN || '';
+const mattermostToken = process.env.MATTERMOST_TOKEN || '';
+const port = parseInt(process.env.FLAMEBOT_PORT || '8000', 10);
+const flameRate = parseFloat(process.env.FLAMEBOT_FLAME_RATE || '0.03');
+const debug = process.env.FLAMEBOT_DEBUG === 'true';
+
+// Validate config
+if (flameRate < 0 || flameRate > 1) {
+  console.error('Flame rate must be between 0 and 1!');
+  process.exit(1);
+}
+if (telegramToken.length === 0) {
+  console.error('Please set TELEGRAM_TOKEN env var');
+  process.exit(1);
+}
+if (mattermostToken.length === 0) {
+  console.error('Please set MATTERMOST_TOKEN env var');
+  process.exit(1);
+}
```

Comment:
Shouldn't you verify the other config values (port, and debug) as well?

Reply:
No, they both have default values if not set.

Full validation isn't necessary either, the program will simply not start and show an error message (e.g. if the type is wrong) :slightly_smiling_face: